static_frame/core/store.py

```diff
@@ -152,6 +185,94 @@ def label_decode(self, label: str) -> tp.Hashable:
         ]
 
 
+class StoreConfigHE(StoreConfig):
+    '''
+    A hash/equals subclass of :obj:`StoreConfig`, permiting usage in a Python set, dictionary, or other contexts where a hashable container is needed.
+
+    To support hashability, label_encoder/label_decoder are ignored
+    '''
+    __slots__ = ('_hash',)
+
+    _not_hashable = ('label_encoder', 'label_decoder')
+
+    _hash: tp.Optional[int]
+
+    def __init__(self, **kwargs): #type: ignore [no-untyped-def]
+        for attr in self._not_hashable:
+            kwargs[attr] = None
+        super().__init__(**kwargs)
+        self._hash = None
+
+    def __eq__(self, other: tp.Any) -> bool:
+        if not isinstance(other, StoreConfig):
+            return False
+        return (
+                self.index_depth == other.index_depth and
+                self.index_name_depth_level == other.index_name_depth_level and
+                self.columns_depth == other.columns_depth and
+                self.columns_name_depth_level == other.columns_name_depth_level and
+                self.columns_select == other.columns_select and
+                self.dtypes == other.dtypes and
+                self.consolidate_blocks == other.consolidate_blocks and
+                self.skip_header == other.skip_header and
+                self.skip_footer == other.skip_footer and
+                self.trim_nadir == other.trim_nadir and
+                self.include_index == other.include_index and
+                self.include_index_name == other.include_index_name and
+                self.include_columns == other.include_columns and
+                self.include_columns_name == other.include_columns_name and
+                self.merge_hierarchical_labels == other.merge_hierarchical_labels and
+                self.read_max_workers == other.read_max_workers and
+                self.read_chunksize == other.read_chunksize and
+                self.write_max_workers == other.write_max_workers and
+                self.write_chunksize == other.write_chunksize
+        )
+
+    def __ne__(self, other: tp.Any) -> bool:
+        return not self.__eq__(other)
+
+    @staticmethod
+    def _hash_depth_specifier(depth_specifier: tp.Optional[DepthLevelSpecifier]) -> tp.Hashable:
+        if depth_specifier is None or isinstance(depth_specifier, int):
+            return depth_specifier
+        return tuple(depth_specifier)
+
+    @staticmethod
+    def _hash_dtypes_specifier(dtypes_specifier: DtypesSpecifier) -> tp.Hashable:
+        if dtypes_specifier is None :
+            return dtypes_specifier
+        if isinstance(dtypes_specifier, dict):
+            return dtypes_specifier.items()
+        if isinstance(dtypes_specifier, list):
+            return tuple(dtypes_specifier)
+        return dtypes_specifier # type: ignore [return-value]
+
+    def __hash__(self) -> int:
+        if self._hash is None:
```

Comment:
This approach of caching the hash I took in SeriesHE, FrameHE for potential performance improvements, but if it simplifies the design, perhaps we do not need it here.

Reply:
I think that it's good to keep, since we potentially have to call something like `.items()` on a dictionary, so making it happen only once lazily seems like the best approach.